internal/guest/storage/scsi/scsi.go

```diff
@@ -163,6 +164,22 @@ func Mount(
 		}
 	}
 
+	// create and symlink block device mount target
+	if config.BlockDev {
+		parent := filepath.Dir(target)
+		log.G(ctx).WithField("parent", parent).Debug("creating parent directory for block device mount")
+		if err := osMkdirAll(parent, 0700); err != nil {
+			return err
+		}
+		defer func() {
+			if err != nil {
+				_ = os.Remove(target)
```

Comment:
the intention was to cleanup `target`, not `parent`. This is probably not needed, since if `Symlink` below fails, we shouldn't have anything to clean up. We never remove `parent` below, it's always the `target` directory. Typically, `parent` will be something like `/run/mounts/scsi/` and the SCSI attachment filesystem will be mounted under that directory. We don't want to remove that.